cibuildwheel/util.py

```diff
@@ -76,40 +77,39 @@ def download(url: str, dest: str) -> None:
         break
 
     try:
-        with open(dest, 'wb') as file:
-            file.write(response.read())
+        dest.write_bytes(response.read())
     finally:
         response.close()
 
 
 class DependencyConstraints:
-    def __init__(self, base_file_path: str):
-        assert os.path.exists(base_file_path)
-        self.base_file_path = os.path.abspath(base_file_path)
+    def __init__(self, base_file_path: Path):
+        assert base_file_path.exists()
+        self.base_file_path = base_file_path.resolve()
 
     @staticmethod
     def with_defaults() -> 'DependencyConstraints':
         return DependencyConstraints(
-            base_file_path=os.path.join(os.path.dirname(__file__), 'resources', 'constraints.txt')
+            base_file_path=Path(__file__).parent / 'resources' / 'constraints.txt'
         )
 
-    def get_for_python_version(self, version: str) -> str:
+    def get_for_python_version(self, version: str) -> Path:
         version_parts = version.split('.')
 
         # try to find a version-specific dependency file e.g. if
         # ./constraints.txt is the base, look for ./constraints-python27.txt
-        base, ext = os.path.splitext(self.base_file_path)
-        specific = base + f'-python{version_parts[0]}{version_parts[1]}'
-        specific_file_path = specific + ext
-        if os.path.exists(specific_file_path):
+        specific_stem = self.base_file_path.stem + f'-python{version_parts[0]}{version_parts[1]}'
+        sepcific_name = specific_stem + self.base_file_path.suffix
```

Comment:
typo! sepcific_name -> specific_name

Reply:
Thanks!
Not sure the `os.path` to `pathlib` conversion is worth it here, but let's be consistent?